### PR TITLE
Introducing modules

### PR DESCRIPTION
**Pull request by nikos-sotiropoulos**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -16,15 +16,22 @@ See the Apache License Version 2.0 for the specific language governing permissio
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
 
     <modelVersion>4.0.0</modelVersion>
-
-    <groupId>com.regnosys.rosetta</groupId>
-    <artifactId>com.regnosys.rosetta.code-generators</artifactId>
-    <packaging>jar</packaging>
+    <groupId>com.regnosys.rosetta.code-generators</groupId>
+    <artifactId>parent</artifactId>
```
Comment (jim-h-wang) [inline pom.xml]: I think the `artefactId` should remain as it was. Did you change it for some technical constraint imposed by Maven?

File: sample/pom.xml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    </parent>
+
+    <artifactId>sample</artifactId>
+    <packaging>jar</packaging>
+    <name>A sample Rosetta code generator</name>
+
+    <dependencies>
+        <dependency>
+            <groupId>com.regnosys.rosetta</groupId>
+            <artifactId>com.regnosys.rosetta</artifactId>
```
Comment (jim-h-wang) [inline sample/pom.xml]: Given you have these dependencies in the parent pom already, they are not needed here, right?

File: sample/pom.xml [hunk 3]
```diff
@@ -0,0 +1,59 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>com.regnosys.rosetta.code-generators</groupId>
+        <artifactId>parent</artifactId>
+        <version>0.0.1</version>
```
Comment (jim-h-wang) [inline sample/pom.xml]: `<artifactId>parent</artifactId>` is this correct?

File: test-helper/pom.xml [hunk 4]
```diff
@@ -0,0 +1,59 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>com.regnosys.rosetta.code-generators</groupId>
+        <artifactId>parent</artifactId>
+        <version>0.0.1</version>
```
Comment (jim-h-wang) [inline test-helper/pom.xml]: same again

File: test-helper/pom.xml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <artifactId>maven-compiler-plugin</artifactId>
+            </plugin>
+<!--            <plugin>-->
+<!--                <artifactId>maven-surefire-plugin</artifactId>-->
+<!--            </plugin>-->
```
Comment (jim-h-wang) [inline test-helper/pom.xml]: Remove commented out code

File: test-helper/src/main/java/com/regnosys/rosetta/generators/test/RosettaExternalGeneratorsModule.java [hunk 6]
```diff
@@ -1,13 +1,18 @@
...
 
+/**
+ * A concrete implementation of Guice's {@link AbstractModule} that is then used to register external generators
+ * See {@link RosettaServerSetup} 
+ *
+ */
 public class RosettaExternalGeneratorsModule extends AbstractModule {
 
-	private final ExternalGeneratorProvider externalGeneratorProvider;
+	private final ExternalGeneratorProvider<?> externalGeneratorProvider;
```
Comment (jim-h-wang) [inline test-helper/src/main/java/com/regnosys/rosetta/generators/test/RosettaExternalGeneratorsModule.java]: Using generics with the <?> seems a little pointless no?

File: test-helper/src/main/java/com/regnosys/rosetta/generators/test/TestHelper.java [hunk 7]
```diff
@@ -8,36 +8,41 @@
...
 import com.regnosys.rosetta.generator.external.AbstractExternalGenerator;
-import com.regnosys.rosetta.generators.sample.SampleCodeGenerator;
 import com.regnosys.rosetta.rosetta.RosettaModel;
 
-public class TestHelper {
+/**
+ * Helper class to parse a rosetta text file into the corresponding root Ecore Rosetta object
+ * 
+ *
+ * @param <T>
```
Comment (jim-h-wang) [inline test-helper/src/main/java/com/regnosys/rosetta/generators/test/TestHelper.java]: better explanation of the param or remove it all together

File: sample/pom.xml (hunk 3 above)
Comment (nikos-sotiropoulos) [reply sample/pom.xml]: yes!

File: sample/pom.xml (hunk 2 above)
Comment (nikos-sotiropoulos) [reply sample/pom.xml]: it is correct

**Pull request merged**